spec/schemas/resource.yaml

```diff
@@ -177,6 +177,13 @@ UserResourceMetadata:
         type: string
       example:
         description: "Human readable description"
+    extensions:
+      type: object
+      description: |
+        User-defined key/value pairs that are mutable and can be used to add extensions.
```

Comment:
I would add, that the extensions will be validated by the CSP. So any value that is not accepted will be rejected on admission.